addons/binding/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/serial/ValloxSerialInterface.java

```diff
@@ -0,0 +1,465 @@
+        telegram[2] = destination;
+        telegram[3] = variable;
+        telegram[4] = value;
+        telegram[5] = calculateChecksum(telegram);
+
+        if (logger.isDebugEnabled()) { // avoid executing the byteToHex if no debug mode is active
+            logger.debug("sending telegram: {} {} {} {} {} {}", Telegram.byteToHex(telegram[0]),
+                    Telegram.byteToHex(telegram[1]), Telegram.byteToHex(telegram[2]), Telegram.byteToHex(telegram[3]),
+                    Telegram.byteToHex(telegram[4]), Telegram.byteToHex(telegram[5]));
+        }
+        serialWrite(telegram);
+    }
+
+    /**
+     * Send one telegram to the master unit.
+     * Example: send(Variable.FAN_SPEED.key, Telegram.convertBackFanSpeed((byte)4));
+     *
+     * @param variable one of the Variable keys in {@link Variable}}
+     * @param value the right binary coded value (see {@link Telegram} for some conversions)
+     * @throws IOException
+     */
+    public void send(byte variable, byte value) throws IOException {
+        send(variable, value, ValloxProtocol.ADDRESS_MASTER);
+    }
+
+    private void serialWrite(byte[] telegram) {
+        try {
+            if (outputStream != null) {
+                outputStream.write(telegram);
+                outputStream.flush();
+            }
+        } catch (IOException e) {
+            String msg = "Exception writing to vallox.";
+            logger.error(msg, e);
+            for (StatusChangeListener l : this.getStatusListener()) {
+                l.statusChanged(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, msg + " " + e.toString());
+            }
+        }
+    }
+
+    static byte calculateChecksum(byte[] pTelegram) {
+        int checksum = 0;
+        for (byte i = 0; i < pTelegram.length - 1; i++) {
+            checksum += pTelegram[i];
+        }
+        return (byte) (checksum % 256);
+    }
+
+    /**
+     * Start listening to telegrams on the serial interface. Stop if requested.
+     */
+    public void startListening() {
+        shutdownListening = false;
+        scheduler.submit(() -> {
+            long sleep = 0;
+            while (!shutdownListening && !Thread.interrupted()) {
+                try {
+                    Telegram telegram = receive();
+                    logger.trace("Received message: {}", telegram);
+                    telegram.updateStore(vallox, valueListener);
+                } catch (MalformedTelegramException e) {
+                    logger.warn("Issue receiving telegram. Discarding.", e);
+                } catch (InvalidRecepientException e) {
+                    logger.debug("Issue receiving telegram. Discarding.", e);
+                } catch (InsufficientDataException e) {
+                    // telegram too short, wait for more data, but ignore
+                    sleep = 200;
+                } catch (IOException e) {
+                    logger.error("Exception reading input stream.", e);
```

Comment:
Is this a bug situation that users should report as an issue if they see this?
Can't we also get here if the connection is simply lost?